plugin/pkg/scheduler/algorithm/predicates/metadata_test.go

```diff
@@ -0,0 +1,324 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicates
+
+import (
+	"fmt"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+	schedulertesting "k8s.io/kubernetes/plugin/pkg/scheduler/testing"
+	"reflect"
+	"testing"
+)
+
+// predicateMetadataEquivalent returns true if the two metadata are equivalent.
+// Note: this function does not compare podRequest.
+func predicateMetadataEquivalent(meta1, meta2 *predicateMetadata) error {
+	if !reflect.DeepEqual(meta1.pod, meta2.pod) {
+		return fmt.Errorf("pods are not the same.")
+	}
+	if meta1.podBestEffort != meta2.podBestEffort {
+		return fmt.Errorf("podBestEfforts are not equal.")
+	}
+	if meta1.serviceAffinityInUse != meta1.serviceAffinityInUse {
+		return fmt.Errorf("serviceAffinityInUses are not equal.")
+	}
+	if len(meta1.podPorts) != len(meta2.podPorts) {
+		return fmt.Errorf("podPorts are not equal.")
+	}
+	for !reflect.DeepEqual(meta1.podPorts, meta2.podPorts) {
+		return fmt.Errorf("podPorts are not equal.")
+	}
+	for k1, v1 := range meta1.matchingAntiAffinityTerms {
```

Comment:
> It might be, but sort interface needs three functions. It should be done for the service affinity as well. We should also be careful that the comparison function almost never generates equal cases, as the order of equal elements in the expected list may not necessarily be the same as what the tests generate and could cause false positives. Given all this, probably sorting and doing DeepEqual may need even more code.

I agree that sort needs 3 function. But those are super trivial function and well separated.

And yes - I agree that in total this might be a bit more code, but it will be better structured and easier to follow.
This function now has 70 lines of code and it's hard to follow it.
And what if we add few more fields to predicateMetadata? We may end up with comparison function having hundreds of line.

If you really don't want to do the sorting (though i think it's easier and less error prone), at the very least please extract comparing those slices to indviidual functions with a TODO to simplify them.